symengine/fields.h

```diff
@@ -0,0 +1,320 @@
+        for (auto &iter : other.get_dict()) {
+            auto t = dict_.lower_bound(iter.first);
+            if (t != dict_.end() and t->first == iter.first) {
+                mp_fdiv_r(t->second, t->second - iter.second, modulo_);
+                if (t->second == 0) {
+                    dict_.erase(t);
+                }
+            } else {
+                integer_class temp;
+                temp = modulo_ - iter.second;
+                mp_fdiv_r(temp, temp, modulo_);
+                dict_.insert(t, {iter.first, temp});
+            }
+        }
+        return static_cast<GaloisFieldDict &>(*this);
+    }
+
+    GaloisFieldDict &operator*=(const GaloisFieldDict &other)
+    {
+        SYMENGINE_ASSERT(modulo_ == other.modulo_);
+        if (dict_.empty())
+            return static_cast<GaloisFieldDict &>(*this);
+
+        auto o_dict = other.get_dict();
+        if (o_dict.empty()) {
+            *this = other;
+            return static_cast<GaloisFieldDict &>(*this);
+        }
+
+        // ! other is a just constant term
+        auto a = o_dict.find(0);
+        if (o_dict.size() == 1
+            and a != o_dict.end()) {
+            for (auto &arg : dict_) {
+                arg.second *= a->second;
+                mp_fdiv_r(arg.second, arg.second, modulo_);
+            }
+            return static_cast<GaloisFieldDict &>(*this);
+        }
+
+        map_uint_mpz dict_out;
+        for (auto &iter : o_dict) {
+            for (auto &it : dict_) {
+                mp_addmul(dict_out[iter.first + it.first], iter.second, it.second);
+                mp_fdiv_r(dict_out[iter.first + it.first],
+                          dict_out[iter.first + it.first], modulo_);
+            }
+        }
+        dict_ = {};
+        for (auto &iter : dict_out) {
+            if (iter.second != integer_class(0))
+                dict_.insert(dict_.end(), {iter.first, iter.second});
+        }
+        return static_cast<GaloisFieldDict &>(*this);
+    }
+
+    friend GaloisFieldDict operator/(const GaloisFieldDict &a, const GaloisFieldDict &b)
+    {
+        GaloisFieldDict c = a;
+        c /= b;
+        return c;
+    }
+
+    GaloisFieldDict &operator/=(const GaloisFieldDict &other)
+    {
+        SYMENGINE_ASSERT(modulo_ == other.modulo_);
+        auto dict_divisor = other.get_dict();
+        if (dict_divisor.empty()) {
+            throw std::runtime_error("ZeroDivisionError");
+        }
+        if (dict_.empty())
+            return static_cast<GaloisFieldDict &>(*this);
+
+        // ! other is a just constant term
+        auto a = dict_divisor.find(0);
+        if (dict_divisor.size() == 1
+            and a != dict_divisor.end()) {
+            integer_class inv;
+            mp_invert(inv, a->second, modulo_);
+            for (auto &iter : dict_) {
+                iter.second *= inv;
+                mp_fdiv_r(iter.second, iter.second, modulo_);
+            }
+            return static_cast<GaloisFieldDict &>(*this);
```

Comment:
Move above 7 lines to a new function `GaloisFieldDict &operator/=(const integer_class &other)`